great_expectations/data_context/util.py

```diff
@@ -542,39 +544,46 @@ def mask_db_url(cls, url: str, use_urlparse: bool = False, **kwargs) -> str:
             url with password masked e.g. "postgresql+psycopg2://username:***@host:65432/database"
         """
         if sa is not None and use_urlparse is False:
-            engine = sa.create_engine(url, **kwargs)
-            return engine.url.__repr__()
+            try:
+                engine = sa.create_engine(url, **kwargs)
+                return engine.url.__repr__()
+            # Account for the edge case where we have SQLAlchemy in our env but haven't installed the appropriate dialect to match the input URL
+            except NoSuchModuleError as e:
+                logger.warning(
+                    f"Something went wrong when trying to use SQLAlchemy to obfuscate URL; could not find dialect {e}"
+                )
         else:
             warnings.warn(
                 "SQLAlchemy is not installed, using urlparse to mask database url password which ignores **kwargs."
             )
+        return cls._mask_db_url_no_sa(url=url)
 
-            # oracle+cx_oracle does not parse well using urlparse, parse as oracle then swap back
-            replace_prefix = None
-            if url.startswith("oracle+cx_oracle"):
-                replace_prefix = {"original": "oracle+cx_oracle", "temporary": "oracle"}
-                url = url.replace(
-                    replace_prefix["original"], replace_prefix["temporary"]
-                )
-
-            parsed_url = urlparse(url)
-
-            # Do not parse sqlite
-            if parsed_url.scheme == "sqlite":
-                return url
+    @classmethod
+    def _mask_db_url_no_sa(cls, url: str) -> str:
+        # oracle+cx_oracle does not parse well using urlparse, parse as oracle then swap back
+        replace_prefix = None
+        if url.startswith("oracle+cx_oracle"):
+            replace_prefix = {"original": "oracle+cx_oracle", "temporary": "oracle"}
```

Comment:
Do you know if there is the same issue with `mysql+pymysql://`? Doesn't matter for now since this looks like code ported as-is to a new func

Reply:
I'm not sure! This was introduced in a change made yesterday and only fails in a weird edge case. Don't have a lot of familiarity here just patching the build.

Let me know if you think anything should be changed (going to merge as is for now just to unblock people).